test/enzyme.jl

```diff
@@ -161,4 +162,59 @@ Enzyme.autodiff(Reverse, f3, Duplicated(copy(A), dA), Duplicated(copy(b1), db1),
 @test dA ≈ dA2 atol=5e-5
 @test db1 ≈ db12
 @test db2 ≈ db22
-=#
\ No newline at end of file
+=#
+
+
+A = rand(n, n);
+dA = zeros(n, n);
+b1 = rand(n);
+function fb(b; alg = LUFactorization())
+    prob = LinearProblem(A, b)
+
+    sol1 = solve(prob, alg)
+
+    sum(sol1.u)
+end
+fb(b1)
+
+manual_jac = map(onehot(b1)) do db
+    y = A \ b1
+    sum(inv(A) * (db - dA*y))
+end |> collect
+@show manual_jac
+
+fd_jac = FiniteDiff.finite_difference_jacobian(fb, b1) |> vec
+@show fd_jac
+
+en_jac = map(onehot(b1)) do db1
+    eres = Enzyme.autodiff(Forward, fb, Duplicated(copy(b1), db1))
+    eres[1]
+end |> collect
+@show en_jac
+
+@test en_jac ≈ manual_jac
+@test en_jac ≈ fd_jac
+
+function fA(A; alg = LUFactorization())
+    prob = LinearProblem(A, b1)
+
+    sol1 = solve(prob, alg)
+
+    sum(sol1.u)
+end
+fA(A)
+
+manual_jac = map(onehot(A)) do dA
+    y = A \ b1
+    sum(inv(A) * (db1 - dA*y))
+end |> collect
+
+fd_jac = FiniteDiff.finite_difference_jacobian(fA, A) |> vec
+
+en_jac = map(onehot(A)) do dA
+    eres = Enzyme.autodiff(Forward, fA, Duplicated(copy(A), dA))
+    eres[1]
+end |> collect
+
+@test_broken en_jac ≈ manual_jac
+@test_broken en_jac ≈ fd_jac
```

Comment:
These tests fail due to numerical imprecision. Not entirely sure why.

Reply:
df/db:
```
manual_jac = [-5.061916901336408, 1.5770609499033128, 5.446411839233853, 0.612648432464526]
fd_jac = [-5.061916947364807, 1.5770610570907593, 5.446411848068237, 0.6126484870910645]
en_jac = [-5.061916901336408, 1.5770609499033128, 5.446411839233853, 0.612648432464526]
```

df/dA:
```
manual_jac = [-1.0254607597283965, 33.43238752846228, 53.515223908678564, 28.42686093118159, 33.84013235610757, 22.569869144629088, 16.00129730157342, 24.207046306464335, 22.715575331910678, 26.035770464186847, 27.970857502536397, 25.5534616182086, 42.01248045607326, 20.02374061097621, 7.20819099154247, 23.21793954106552]
fd_jac = [-26.272539049386978, 8.185315012931824, 28.268145620822906, 3.1797885596752167, 8.593059450387955, -2.677203118801117, -9.245775252580643, -1.0400259792804718, -2.5314970314502716, 0.7886982262134552, 2.7237851321697235, 0.3063894212245941, 16.765405893325806, -5.22333163022995, -18.038884073495865, -2.029132694005966]
en_jac = [-26.272532992576195, 8.185315295614476, 28.268151675830794, 3.1797886983337977, 8.593060123259754, -2.6772030882187066, -9.24577493127438, -1.0400259263834648, -2.531496900937129, 0.7886982313390366, 2.7237852696885847, 0.30638938536080607, 16.76540822322545, -5.223331621871595, -18.038881241305333, -2.0291326917822765]
```

Reply:
That's an expected difference due to summation order.

Reply:
addressed in 31745cc